cabot/metricsapp/tasks.py

```diff
@@ -0,0 +1,170 @@
+from django.core.mail import send_mail
+from cabot.metricsapp.api import get_dashboard_info, get_updated_datetime, get_panel_info, \
+    create_generic_templating_dict, template_response, get_es_status_check_fields, get_series_ids
+from cabot.metricsapp.defs import GRAFANA_SYNC_TIMEDELTA_MINUTES
+from cabot.metricsapp.models import MetricsStatusCheckBase, ElasticsearchStatusCheck, GrafanaDataSource
+
+
+celery = Celery(__name__)
+celery.config_from_object(settings)
+logger = logging.getLogger(__name__)
+
+
+def _dashboard_deleted_handler(check, panel):
+    """
+    When the dashboard is deleted in Grafana, deactivate the check and send an email
+    :param check: the status check
+    :param panel: the GrafanaPanel object
+    :return None
+    """
+    check.active = False
+    check.save()
+    send_grafana_sync_email.apply_async([str(check.created_by.email)],
+                                        '{}\n\n'
+                                        'Dashboard {} has been deleted, so the check {} has been deactivated. '
+                                        'If you would like to keep the check, re-enable it with auto_sync: False.'
+                                        .format(check.get_url_for_check(),
+                                                panel.dashboard_uri.split('/')[1],
+                                                check.name),
+                                        check.name)
+
+
+def _panel_deleted_handler(check, panel, dashboard_meta):
+    """
+    When a check's panel is deleted in Grafana, deactivate the check and send an email
+    :param check: the status check
+    :param panel: the GrafanaPanel object
+    :param dashboard_meta: "meta" section of Grafana dashboard api response
+    :return None
+    """
+    check.active = False
+    check.save()
+    send_grafana_sync_email.apply_async([str(check.created_by.email), str(dashboard_meta['createdBy']),
+                                         str(dashboard_meta['updatedBy'])],
+                                        '{}\n\n'
+                                        'Panel {} in dashboard {} has been deleted, so the check {} has been '
+                                        'deactivated. If you would like to keep the check, re-enable it with '
+                                        'auto_sync: False.'.format(check.get_url_for_check(),
+                                                                   panel.panel_id,
+                                                                   panel.dashboard_uri.split('/')[1],
+                                                                   check.name),
+                                        check.name)
+
+
+@task(ignore_result=True)
+def sync_all_grafana_checks():
+    """Task to sync all status checks with auto_sync set to their Grafana dashbaords"""
+    sync_time = datetime.now()
+    for check in MetricsStatusCheckBase.objects.filter(auto_sync=True).filter(active=True):
```

Comment:
This will fail when we have a new MetricsStatusCheck which is not a grafana type, but has `auto_sync` = True. Ideally we should filter only for grafana checks.